src/serve.rs

```diff
@@ -126,10 +133,26 @@ impl ServeSystem {
         ));
         let router = router(state, cfg.clone())?;
         let addr = (cfg.address, cfg.port).into();
-        let server = Server::bind(&addr)
-            .serve(router.into_make_service())
-            .with_graceful_shutdown(shutdown_fut);
 
+        let mut http_server: Option<_> = None;
```

Comment:
I am not a fan of this duplication … and I don't fully understand what's going on with the shutdown. I will try to see if we can improve this. Never worked with axum before, so I need to better understand this.

Reply:
Just to be clear, I wouldn't consider this blocker. Just room for improvement.

Reply:
Ok, I pushed some changes, consolidating the changes/differences into some function. Seems like the types are a bit complex to get them into a box or something.

I will go ahead and try this locally (non-TLS), maybe you can do the same (with TLS). It that works fine for both of us, then I think this part of resolved.